### PR TITLE
renaming/cleanup for sharepoint perms pt.1

### PR DESCRIPTION
**Pull request by ryanfkeepers**
No logic changes, just renaming and minor cleanup.

PRs to follow:
1. collect various maps in onedrive collections into a single cache.
2. logic changes and tests to produce sharepoint permissions backup/restore e2e.
3. extend permission identity-type retention.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3135

#### Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/internal/connector/onedrive/permission.go [hunk 1]
```diff
@@ -207,12 +209,12 @@ func UpdatePermissions(
 
 		pbody.SetRecipients([]models.DriveRecipientable{rec})
 
-		np, err := service.Client().DrivesById(driveID).ItemsById(itemID).Invite().Post(ctx, pbody, nil)
+		newPerm, err := api.PostItemPermissionUpdate(ctx, service, driveID, itemID, pbody)
 		if err != nil {
-			return graph.Wrap(ctx, err, "setting permissions")
+			return err
```
Comment (ashmrtn) [inline src/internal/connector/onedrive/permission.go:214]: will we get nice stack trace info that shows this is where `PostItemPermissionUpdate` was called if we don't wrap the error here? I haven't really looked through all of the stack trace generation code to know

File: src/internal/connector/onedrive/restore.go [hunk 2]
```diff
@@ -157,17 +157,17 @@ func RestoreCollection(
 		return metrics, clues.Wrap(err, "creating drive path").WithClues(ctx)
 	}
 
-	if rootIDCache == nil {
-		rootIDCache = map[string]string{}
+	if rootFolderIDCache == nil {
+		rootFolderIDCache = map[string]string{}
```
Comment (ashmrtn) [inline src/internal/connector/onedrive/restore.go:161]: I still feel like this is a code smell. It seems like either we should be passing it in all the time and error out if it's not or we should never pass it in and create it in this function

**Review by ashmrtn — APPROVED** [review]

File: src/internal/connector/onedrive/restore.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/restore.go:161]: It is a smell, but it gets cleaned up in the next PR.

File: src/internal/connector/onedrive/permission.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/permission.go:214]: Stack trace generation is attached to any clues wrap or stack call.  So we can stack here, too, if wanted.

**Pull request merged**